### ISSUE TITLE
export - import

### ISSUE THREAD

**Issue by pofider** [jsreport/jsreport#221]
extension `jsreport-export-import` capable of exporting/importing all the templates,scripts.. into/from file or stream. It would be useful for transferring templates between dev/test/prod environments, possibly automated with API.

The options for export would be:
- entity types to export (default all)
- export target
  - file stream (default)
  - target environment (user can optionally specify url and credentials to the target environment) (this has probably low priority)

The import options would be:
- source file
- drop - clean restore (drop whole entity type collection before inserting)
- entity types to import (default all)

All of this would use our node document store API, so it would work universally across all kind of stores like fs, mongo, sql... This implies it would also provide another important feature which is migration across different data stores.

**Comment by zam6ak** [comment]
@pofider I think this look good, but I would not mix backup/restore functionality with import/export.
IMHO, backup/restore should be perhaps npm run {script} kind of activity and not something a user would do via UI (opportunity for users to mess up stuff)...

As for import/export it would be nice to support exporting and importing  "a package".
A package could contain a template, few assets, a script or two. Something that to end user has meaning as a whole. It could be as simple as creating a ZIP of several selected export items.

I also agree with low priority for export/import to target environment.

**Comment by pofider** [comment]
@zam6ak Ah ok, you are rather interested in export/import particular objects than whole collections. This make sens, but we gonna have to provide an entity tree in which you can select these items through check boxes.

**Comment by zam6ak** [comment]
Great ! I will test as soon as new release is out

**Comment by pofider** [comment]
The preview can be installed like this:

```
npm install jsreport-import-export --production
npm install jsreport-studio@1.2.3 --production
```

This will be part of the next jsreport release, which should come after the Christmas .

**Comment by pofider** [comment]
This is now released in jsreport@1.3.0. We can solve the coming problems in separate issues.
